core/modules/layout/includes/block.copy.inc

```diff
@@ -0,0 +1,121 @@
+<?php
+
+/**
+ *  BlockCopy extends Block
+ *
+ *  This class allows us to create blocks which are copies of other blocks.
+ */
+class BlockCopy extends Block {
+  /**
+   * {@inheritdoc}
+   */
+  function __construct($plugin_name, array $data) {
+    parent::__construct($plugin_name, $data);
+
+  }
+
+  /**
+   *  Sets title text on draggable block panel in Layout builder.
+   */
+  function getAdminTitle() {
+    $parent_block = $this->getParentBlock();
+    $parent_block_info = $this->getParentBlockInfo($parent_block);
+    return t('Copy of %block', array('%block' => $parent_block_info['info']));
+  }
+
+  /**
+   *  Sets block subject on block view.
+   */
+  function getTitle() {
+    $parent_block = $this->getParentBlock();
+    return isset($parent_block->settings['title']) ? check_plain($this->parent_block['title']) : '';
+  }
+
+  /**
+   * Returns the rendered content of this block.
+   *
+   * @return string
+   */
+  function getContent() {
+    if (!empty($this->settings['source_block'])) {
+      $parent_block = $this->getParentBlock();
+      return $parent_block->getContent();
+    }
+  }
+
+  /**
+   *  Builds the block's configuration form.
+   */
+  function form(&$form, &$form_state) {
+    parent::form($form, $form_state);
+    // Hide the title display.
+    $form['title_display']['#access'] = FALSE;
+
+    $all_block_info = layout_get_block_info();
+    if (!isset($this->settings['source_block'])) {
+      $blocks = array();
+      $usage = layout_get_block_usage();
+      foreach ($usage as $module => $block) {
+        foreach ($block as $block_delta => $layout_blocks) {
+          foreach ($layout_blocks as $layout_name => $block_positions) {
```

Comment:
I think it would be preferred to first select a Layout in a select list and then select the Region + Block in a second select list. Otherwise this list will easily become hundreds of blocks long.